Fixes #58: Adds a JS testing framework and other JS goodies

filmaj opened the pull request:
I needed to make a few big changes to add JS testing. Additionally, this PR includes many tweaks to the documentation - so feedback is much appreciated 😄 

# Summary of Changes

1. By far the biggest change is that I moved the JS dependencies we currently have (jquery, moment.js, Chart.js, d3.js) into this repo. I needed to do this so that the test runner is able to load them into our target JS environments (Chrome, in our case, but potentially we could add more in the future), since the main `charts.js` source file we author directly depends on them and assumes they are available globally.
    - If y'all don't like this change, I can look into seeing how viable mocking out all our JS dependencies would be. However, I expect this to be a tedious task, and if we ever add new JS dependencies (or update existing ones), we'd have to also update the mocks. 😒 
1. Tweaked `CONTRIBUTING.md` documentation to explain where to look for info on how to validate (i.e. run the tests) the web app component.
1. Clarified the Getting Started section of the top-level `README.md` to explain what the updater and docs subdirectory / components are and linking to their relevant READMEs.
1. Added a `README.md` to the `docs/` component, explaining (in so far as I understand) how the web app works, how it is structured, and how to run tests (and in the hopefully-near-future, linter).
1. Tweaked jekyll `_config.yml` to exclude the extra node.js bits, like dependency directory and config files.
1. Added basic smoke tests for `charts.js` - nothing more than checking for the existence of our main `createChart` functions. Hopefully, with every bug we identify therein, or with every new JS feature we add, we can also add tests now.

# Try It Out

Ensure you have Google Chrome installed (and updated). Ensure you have a recent version of [node.js](https://nodejs.org) available. Pull this branch down, then:

    $ cd docs
    $ npm install # install dependencies
    $ npm test # should exit with code 0 and print out friendly green text saying 4 tests passed

Want to see what a failing test looks like? Edit `docs/spec/charts.js`, and inside one of the test blocks (encapsulated via the `it()` function), add `expect(false).toBe(true);` - then rerun the tests via `npm test`. Oh no! Angry red text and error stack traces everywhere!

larsxschneider commented on docs/README.md:
```diff
@@ -0,0 +1,46 @@
+
+### Linting
+
+    $ npm run lint
```
The linter complains it can't find a config file. I guess this is expected?

filmaj replied:
Ah yes, I have it ready to go but I was going to wait until the tests were merged before adding all the various bits. I will remove these instructions for now until I put together the full additions for linting. I'll also rebase with latest master.

larsxschneider commented:
Plus, don't worry about the conflict. I can fix it while merging.

filmaj commented:
I've rebased and removed the linter instructions. Once this is merged, I've got all the parts for the linter ready to go - but I'll issue a separate PR for that.
